lbz/authz/collector.py

```diff
@@ -0,0 +1,28 @@
+from lbz.misc import Singleton
+
+
+class AuthzCollector(metaclass=Singleton):
+    # TODO: Add uni tests
+    def __init__(
+        self,
+    ) -> None:
+        self.possible_permissions: list = []
+        self.resource = ""
+        self.guest_permissions: dict = {}
+
+    def set_resource(self, resource_name: str) -> None:
+        self.resource = resource_name
```

Comment:
If we want to store here the name, it should be `self.resource_name` (the same for the name of the method) just to not mislead anyone that this is a full `Resource` object.